### PR TITLE
feat: botframework http adapter

### PR DESCRIPTION
**Pull request by joshgummersall**
Fixes https://github.com/microsoft/botbuilder-js/issues/3420

Also includes:
- standard lib `delay` function
- consolidated sinon versions
- fix API calls to accept `Partial<Activity>` instead of obfuscated `as Activity` invocation pattern

### CODE REVIEW

File: libraries/botframework-schema/src/index.ts [hunk 1]
```diff
@@ -1947,6 +1947,7 @@ export enum ActivityTypes {
     Suggestion = 'suggestion',
     Trace = 'trace',
     Handoff = 'handoff',
+    Delay = 'delay',
```
Comment (stevengum) [inline libraries/botframework-schema/src/index.ts:1950]: `'delay'` is a "fake" activity type and is only supported by the BotFrameworkAdapter. Can you roll this back?

For reference, the official Activity types are defined in the [Bot Framework Activity specification](https://github.com/microsoft/botframework-obi/blob/main/protocols/botframework-activity/botframework-activity.md).

File: libraries/botframework-connector/src/connectorApi/operations/conversations.ts [hunk 2]
```diff
@@ -124,21 +124,21 @@ export class Conversations {
    * @param [options] The optional parameters
    * @returns Promise<Models.ConversationsSendToConversationResponse>
    */
-  sendToConversation(conversationId: string, activity: Models.Activity, options?: msRest.RequestOptionsBase): Promise<Models.ConversationsSendToConversationResponse>;
+  sendToConversation(conversationId: string, activity: Partial<Models.Activity>, options?: msRest.RequestOptionsBase): Promise<Models.ConversationsSendToConversationResponse>;
```
Comment (stevengum) [inline libraries/botframework-connector/src/connectorApi/operations/conversations.ts:127]: This seems fine to me! But I want to bring up that the Activity contract should correctly define all the necessary information for a channel to properly process the Activity. If this holds, then changing to a `Partial` will open the door to customers sending malformed requests. Particularly customers using the `botframework-connector` library whom aren't using `botbuilder-core`.

**Review by stevengum — CHANGES_REQUESTED** [review]
The comments I'd like to see addressed are the BotFrameworkHttpAdapter typo and the ActivityTypes.Delay.

File: libraries/botframework-connector/src/connectorApi/operations/conversations.ts (hunk 2 above)
Comment (joshgummersall) [reply libraries/botframework-connector/src/connectorApi/operations/conversations.ts:127]: I agree with you in principle, though there are a few concerns I would describe as more pragmatic:

- The SDK currently makes no attempt to perform any validation short of `as Activity`
- Duplicating validation that already exists in the API layer couples our source code with the API which means updates have to happen in lock step

It's not an ideal change, but all the casting that exists currently is worse in my opinion.

**Review by stevengum — APPROVED** [review]

**Pull request merged**